### ISSUE TITLE
default http client has way too low MaxIdleConnsPerHost

### ISSUE THREAD

**Issue by weinong** [Azure/azure-sdk-for-go#19192]
### Bug Report

- What happened?
  - the [default http client](https://github.com/Azure/azure-sdk-for-go/blob/b386adc15c2e60e3812956f68d244b608330c479/sdk/azcore/runtime/transport_default_http_client.go#L26) that azure sdk uses has `MaxIdleConns: 100`. However, as it doesn't change `MaxIdleConnsPerHost`, the default is 2. This means for 100 connections it opens, it also closes 98 of them. [ref](https://github.com/OJ/gobuster/issues/127) and [ref](https://gitlab.com/gitlab-org/gitlab-pages/-/merge_requests/274). It's particularly bad on Azure due to the infamous SNAT port exhaustion.
- What did you expect or want to happen?
  - change `MaxIdleConnsPerHost` to `100` as well

**Comment by jhendrixMSFT** [comment]
I assume that on the AKS side there was some testing to arrive at this number.  Do you have any details on that?  Given that this would change the in-box default, I'd like to make sure there aren't any unwanted side effects.

**Comment by weinong** [comment]
the reason I raised this issue was because I was fixing a connection leak bug and found some references (linked in the issue) that suggests to update the default `MaxIdleConnsPerHost`. The reasoning makes sense.
But I did some digging today and found I didn't see any change regardless `MaxIdleConnsPerHost` is set or not. Upon digging further, I found it only happened in HTTP 1.1, but not in HTTP/2. Since the SDK defaults to HTTP/2 with Azure Resource Manager, I think we can close it.
